app/controllers/subsidy/applications/edit.js

```diff
@@ -2,9 +2,18 @@ import Controller from '@ember/controller';
 import { inject as service } from '@ember/service';
 import { task } from 'ember-concurrency';
 import fetch from 'fetch';
+import { action } from '@ember/object';
+import { saveAs } from 'file-saver';
+import JSZip from 'jszip';
```

Comment:
But related to that. It seems to cover the same function as the PR I linked, why reimplement that here?

Reply:
I believe the PR's functionality works on a per file-upload input base, in here we wanted to combine all the files in from the form in a zip. I tried re-using some of the other code with client-zip but that seemed more complex so I just kept this, but I can take another look

Reply:
But how often does it happen that there are multiple different file upload fields in the same form?

We opted out of this in ember-submission-form-fields because it made things more complex for little value. Most forms (in toezicht) have only a single file upload and/or remote url field, so this would have saved 1 click. Not sure about the subsidy forms, but still.

Reply:
For subsidies it happens quite often actually, fietsinfrastructuur for example has 2 steps with 5+ file uploads. Plus it's handy to have the button at the top of the page

Reply:
In any case, I would try to split off the batch download feature from the pdf printing work. It feels unrelated and I think we can come up with a better solution.

Reply:
> For subsidies it happens quite often actually, fietsinfrastructuur for example has 2 steps with 5+ file uploads. Plus it's handy to have the button at the top of the page

Ah, I see, I missed this reply. Yea that makes sense then. I still don't like the way it's implemented though :sweat_smile: . Depending on implementation details of the forms addon feels very frail.

Reply:
I refactored to use client-zip with triggerZipDownload instead of jszip and file-saver. I hope this will do and we can merge it in together :)